### DISCUSSION TITLE
How far can I push render quality on Pi 4?

### DISCUSSION THREAD

**Discussion by jbienz** [FormerLurker/Octolapse#706] [Q&A] [answered]
I followed the DSLR guide in the Wiki and on Youtube. Both were fantastic and I was up and running right away. THANK YOU @FormerLurker!

My question is on maxing out the render settings. Here is my hardware configuration:

- **Compute:** Raspberry Pi 4
- **RAM:** 8GB
- **Storage:** 128 GB U3 extreme
- **Camera:** EOS Rebel XSI
- **Snapshot Resolution:** 4272 x 2848

My questions:

1. With the above config, should I still avoid H.264? I see mention of running out of RAM, but I don't know if that's for older PI versions with less RAM or if I'll still run out on 8GB.
2. With the snapshot resolution and hardware config above, what Bitrate should I max out at? The guide suggests 10 - 20k, but it seems the average bitrate for 4k video is 35 - 45k. Would it be insane for me to set the libxvid render bitrate to 40k with this setup? I'm not overly concerned about storage. I have over 100 GB free and I will offload to my PC after each render.
3. Looks like the Pi 4 is quad core. Just confirming it's safe for me to give OctoLapse 3 cores? I highly doubt I'll ever do back-to-back prints so I think it should be safe, but I just don't want to make OctoPi unresponsive.

Thank you again so much @FormerLurker. Your project has added so much joy to picking up the hobby of 3D printing!

**Comment by FormerLurker** [comment, marked as answer]
> Thank you again so much @FormerLurker. Your project has added so much joy to picking up the hobby of 3D printing!

No problem, thank you!

> With the above config, should I still avoid H.264? I see mention of running out of RAM, but I don't know if that's for older PI versions with less RAM or if I'll still run out on 8GB.

Well, it all depends how long the video is.  If you want, you can use the archive option within the render profile to save your snapshots within Octolapse.  That way you can test rendering with various settings via the 'Files and Videos' button on the tab.  The way I do it usually is to completely disable rendering (this automatically archives the images), then I go in and render with various other profiles.  If it fails, all you've lost is time (though Ocotlapse now will recover from rendering problems and allow you to try again if it crashes).

> With the snapshot resolution and hardware config above, what Bitrate should I max out at? The guide suggests 10 - 20k, but it seems the average bitrate for 4k video is 35 - 45k. Would it be insane for me to set the libxvid render bitrate to 40k with this setup? I'm not overly concerned about storage. I have over 100 GB free and I will offload to my PC after each render.

Yeah, I would crank it up.  Your images are more than 4k, so you will need to go higher to get really good quality.  Again, I would test them out.  

> Looks like the Pi 4 is quad core. Just confirming it's safe for me to give OctoLapse 3 cores? I highly doubt I'll ever do back-to-back prints so I think it should be safe, but I just don't want to make OctoPi unresponsive.

Yeah, I set the cores to 3 on my Pi3B+.  OctoPrint only runs on a single core (though there are other background processes like mjpgstreamer).  If you have any problems using 3 cores, which I have not, 2 should be totally safe.  If you look at a resource monitor you will see that typically only a single core is taxed, and Mjpgstreamer uses a fraction of a single core.  If you watch it while rendering with 3 cores, 3 cores will be 100% maxed.  If your cooling and PSU are good, that shouldn't be a problem.

Hope that helps!
